### ISSUE TITLE
Incorrect version 1.15.0 available as latest Go module

### ISSUE THREAD

**Issue by paoloambrosio-skyuk** [argoproj/argo-events#1463]
**Describe the bug**

There seems to be a [version `v1.15.0` (Oct 12, 2021) available](https://pkg.go.dev/github.com/argoproj/argo-events?tab=versions) when using `go mod` to bring in the `github.com/argoproj/argo-events` dependency, despite the latest tag being currently `v1.5.4` (Dec 10, 2021).

**To Reproduce**

```
$ go get github.com/argoproj/argo-events
go: downloading github.com/argoproj/argo-events v1.15.0
```

**Expected behavior**

Version from the latest tag (currently `v1.5.4`) should be downloaded and used.

**Screenshots**

N/A

**Environment (please complete the following information):**

 - Go: 1.17 (but it should be irrelevant)

**Additional context**

N/A

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by whynowy** [comment]
@paoloambrosio-skyuk - thanks for reporting the issue. That version was published accidentally, I retracted it, can you retry to see if you still see that?

**Comment by paoloambrosio-skyuk** [comment]
@whynowy thanks for looking into this.

Unfortunately, it doesn't seem to have worked. `go get` seems to be able to still find it:
```
~ go clean -modcache
~ rm -fr go
~ go get github.com/argoproj/argo-events
go: downloading github.com/argoproj/argo-events v1.15.0
```

Not sure if I'm doing the right thing (I'm new to Go), but [`pkg.go.dev`](https://pkg.go.dev/github.com/argoproj/argo-events?tab=versions) still displays that version as well.

It is not impacting us at the moment, since we have fixed the version to the latest one released. If `retract` is considered only when publishing a new release, I'd day that it can definitely wait till `v1.5.5` is published.

**Comment by whynowy** [comment]
@paoloambrosio-skyuk - thanks for your feedback! This is also a new case for us, I'm not quite sure if release a new revise version (v1.5.5) make it retract a future mistaken version (v1.15.0), but we will try to figure out.

**Comment by whynowy** [comment]
@paoloambrosio-skyuk - should have been fixed.

**Comment by paoloambrosio-skyuk** [comment]
Yes, it's fixed! Now it downloads [v1.5.6](https://pkg.go.dev/github.com/argoproj/argo-events@v1.5.6) 👍🏻
